sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/UnivocityParser.scala

```diff
@@ -52,7 +52,8 @@ class UnivocityParser(
     requiredSchema: StructType,
     val options: CSVOptions,
     filters: Seq[Filter]) extends Logging {
-  require(requiredSchema.toSet.subsetOf(dataSchema.toSet),
+  require(requiredSchema.toSet.subsetOf(
```

Comment:
Is this specific to CSV? May it be safe to remove this check?

Reply:
I have searched Spark's Codebase. At present, this specific to CSV.
From a logical perspective, deleting this logic is also a solution.
So let us delete it first?

Reply:
Let's try

Reply:
cc @HyukjinKwon

Reply:
> Let's try

+1

Reply:
Done.